client/me/notification-settings/reader-subscriptions/index.jsx

```diff
@@ -29,209 +30,205 @@ import observe from 'lib/mixins/data-observe';
+								<option value="6">{ this.props.translate( 'Saturday' ) }</option>
+							</FormSelect>
+
+							<FormSelect
+								disabled={ this.getDisabledState() }
+								id="subscription_delivery_hour"
+								name="subscription_delivery_hour"
+								onChange={ this.updateSetting }
+								onFocus={ this.recordFocusEvent( 'Email Delivery Window Time' ) }
+								value={ this.getSetting( 'subscription_delivery_hour' ) }
+							>
+								<option value="0">{ this.getDeliveryHourLabel( 0 ) }</option>
+								<option value="2">{ this.getDeliveryHourLabel( 2 ) }</option>
+								<option value="4">{ this.getDeliveryHourLabel( 4 ) }</option>
+								<option value="6">{ this.getDeliveryHourLabel( 6 ) }</option>
+								<option value="8">{ this.getDeliveryHourLabel( 8 ) }</option>
+								<option value="10">{ this.getDeliveryHourLabel( 10 ) }</option>
+								<option value="12">{ this.getDeliveryHourLabel( 12 ) }</option>
+								<option value="14">{ this.getDeliveryHourLabel( 14 ) }</option>
+								<option value="16">{ this.getDeliveryHourLabel( 16 ) }</option>
+								<option value="18">{ this.getDeliveryHourLabel( 18 ) }</option>
+								<option value="20">{ this.getDeliveryHourLabel( 20 ) }</option>
+								<option value="22">{ this.getDeliveryHourLabel( 22 ) }</option>
+							</FormSelect>
+
+							<FormSettingExplanation>
+								{ this.props.translate(
+									'When choosing daily or weekly email delivery, which time of day would you prefer?'
+								) }
+							</FormSettingExplanation>
+						</FormFieldset>
+
+						<FormFieldset>
+							<FormLegend>{ this.props.translate( 'Block Emails' ) }</FormLegend>
+							<FormLabel>
+								<FormCheckbox
+									checked={ this.getSetting( 'subscription_delivery_email_blocked' ) }
+									disabled={ this.getDisabledState() }
+									id="subscription_delivery_email_blocked"
+									name="subscription_delivery_email_blocked"
+									onChange={ this.toggleSetting }
+									onClick={ this.recordCheckboxEvent( 'Block All Notification Emails' ) }
+								/>
+								<span>
+									{ this.props.translate(
+										'Block all email updates from blogs you’re following on WordPress.com'
 									) }
-								</FormButton>
-							</form>
-						</Card>
-					</Main>
-				);
-			},
-		} )
-	)
-);
+								</span>
+							</FormLabel>
+						</FormFieldset>
+
+						<FormButton
+							isSubmitting={ this.state.submittingForm }
+							disabled={ this.getDisabledState() }
+							onClick={ this.recordClickEvent( 'Save Notification Settings Button' ) }
+						>
+							{ this.state.submittingForm ? (
+								this.props.translate( 'Saving…' )
+							) : (
+								this.props.translate( 'Save Notification Settings' )
+							) }
+						</FormButton>
+					</form>
+				</Card>
+			</Main>
+		);
+	},
+} );
+
+const enhance = flow( protectForm, localize );
+
+export default enhance( NotificationSubscriptions );
```

Comment:
Hmm, can we inline `enhance`, i.e.

```js
export default flow( protectForm, localize )( NotificationSubscriptions );
```

?

(Reads more fluent to me -- that might be subjective, admittedly.)

Reply:
Sure thing :)

My reasoning for this approach is that it starts to become quite noisy when the composed functions force the `flow` to grow beyond a one liner - I think connect is the only culprit at the moment. Even though we don't have `connect` in use here I've been trying lately to use patterns without being subjective - removing the 'if we use this, then do this, otherwise use this pattern' thinking.

All that said though, inlining as such...:

```js
export default flow( 
  protectForm, 
  localize,
  connect(
    state => ( {
      // lots and
      // lots and
      // lots of things
    } )
  )
)( NotificationSubscriptions );
```

...Still totally works for me `¯\_(ツ)_/¯`

Reply:
I've seen some people define `const connectComponent =  connect(...)` in cases like that. Granted, it's just shuffling around what to inline and what to indirect :slightly_smiling_face: